test/extended/crc/ux/installer/installer_windows.go

```diff
@@ -4,25 +4,136 @@ package installer
 )
 
-type handler struct {
-	currentUserPassword *string
-	installerPath       *string
+var installFlow = []element{
+	{"Next", elementClickTime, ux.BUTTON},
+	{"I accept the terms in the License Agreement", elementClickTime, ux.CHECKBOX},
+	{"Next", elementClickTime, ux.BUTTON},
+	{"Next", elementClickTime, ux.BUTTON},
+	{"Install", installationTime, ux.BUTTON},
+	{"Finish", elementClickTime, ux.BUTTON}}
+
+var rebootButton = element{"Yes", elementClickTime, ux.BUTTON}
+
+type element struct {
+	id          string
+	delay       time.Duration
+	elementType string
+}
+
+type gowinxHandler struct {
+	CurrentUserPassword *string
+	InstallerPath       *string
+	Finished            chan bool
 }
 
-func NewInstaller(currentUserPassword *string, installerPath *string) Installer {
+func NewInstaller(currentUserPassword, installerPath *string) Installer {
 	// TODO check parameters as they are mandatory otherwise exit
-	return handler{
-		currentUserPassword: currentUserPassword,
-		installerPath:       installerPath}
+	return gowinxHandler{
+		CurrentUserPassword: currentUserPassword,
+		InstallerPath:       installerPath,
+		Finished:            make(chan bool)}
 
 }
 
 func RequiredResourcesPath() (string, error) {
 	return "", nil
 }
 
-func (h handler) Install() error {
-	return fmt.Errorf("not implemented yet")
+func (g gowinxHandler) Install() error {
+	// Initialize context
+	ux.Initialize()
+	if err := runInstaller(*g.InstallerPath, g.Finished); err != nil {
+		return err
+	}
+	time.Sleep(installerStartTime)
+	for _, action := range installFlow {
+		// delay to get window as active
+		// need to get window everytime as its layout is changing
+		installer, err := ux.GetActiveElement(installerWindowTitle, ux.WINDOW)
+		if err != nil {
+			return err
+		}
+		element, err := installer.GetElement(action.id, action.elementType)
+		if err != nil {
+			err = fmt.Errorf("error getting %s with error %v", action.id, err)
+			logging.Error(err)
+			return err
+		}
+		if err := element.Click(); err != nil {
+			err = fmt.Errorf("error clicking %s with error %v", action.id, err)
+			logging.Error(err)
+			return err
+		}
+		// Delay after action
+		time.Sleep(action.delay)
+	}
+	// TODO which should be executed from a new cmd (to ensure ENVs)
+	// Finalize context
+	ux.Finalize()
+	return nil
+}
+
+func (g gowinxHandler) RebootRequired() error {
+	ux.Initialize()
+	installer, err := ux.GetActiveElement(installerWindowTitle, ux.WINDOW)
+	if err != nil {
+		return err
+	}
+	element, err := installer.GetElement(rebootButton.id, rebootButton.elementType)
+	if err != nil {
+		return err
+	}
+	if err := element.Click(); err != nil {
+		err = fmt.Errorf("error clicking %s with error %v", rebootButton.id, err)
+		logging.Error(err)
+		return err
+	}
+	<-g.Finished
+	ux.Finalize()
+	return nil
+}
+
+func runInstaller(installerPath string, finished chan bool) error {
+	// cmd := exec.Command("powershell.exe", "-c", "msiexec.exe", "/i", installerPath, "/qf")
```

Comment:
Is this needed?

Reply:
Ouch, totally forgot about it..definitely forgot cleanup code and  I will give a try without the subroutine as probably it is not required, but it was changed due to some issues I got to pick the elements which in the end was not related to this but with needed for an active X session. (Same applies to next comment)

Reply:
removed dead code + refactor removing the async part